### ISSUE TITLE
About face crop and cluster

### ISSUE THREAD

**Issue by joaoluis2560** [tomas-gajarsky/facetorch#69]
Hi Tomas, thanks for the repo! I have a question, is there any way to save only the detected face? I didn't find this option.

And would you have any suggestions for the clustering task, any approaches that might work better?

Thanks!

**Comment by tomas-gajarsky** [comment]
Hi there,

Thanks for your interest in facetorch and for reaching out with your questions!

### Saving Detected Faces
Regarding saving only the detected faces, while facetorch doesn't have a built-in function for this, you can certainly achieve it using the location details of each detected face. Each face location is represented by a [`Location`](https://tomas-gajarsky.github.io/facetorch/facetorch/datastruct.html#facetorch.datastruct.Location) object with `x1`, `x2`, `y1`, and `y2` coordinates, among others. Here’s how you can use this information:

1. Access the location of each face with `{face.indx: (face.loc.x1, face.loc.y1, face.loc.x2, face.loc.y2) for face in response.faces}`.
2. Use these coordinates to crop the face from the original image. You can use an image processing library like PIL (Python Imaging Library) for cropping.

Example using PIL:

```python
from PIL import Image

# Assuming 'original_image' is your loaded image and 'face_locations' is the dictionary with coordinates
for indx, (x1, y1, x2, y2) in face_locations.items():
    face_image = original_image.crop((x1, y1, x2, y2))
    face_image.save(f"detected_face_{indx}.jpg")
```

This will save each detected face as a separate image file.

### Clustering Task
For your clustering needs, I suggest utilizing the [embeddings](https://github.com/tomas-gajarsky/facetorch?tab=readme-ov-file#facial-representation-learning-embed) produced by facetorch, which you can access via `face.preds["embed"].logits`. These embeddings can serve as inputs for various clustering algorithms, providing a means to group similar faces together. This approach allows you to leverage the deep learning features captured by the neural network for potentially more effective clustering.

Hope this helps! If you have further questions or need more assistance, feel free to ask.

Best,
Tomas

**Comment by joaoluis2560** [comment]
Awesome, tks Tomas
